programs/blockrunners/src/instructions/generate_player_path.rs

```diff
@@ -0,0 +1,29 @@
+use anchor_lang::prelude::*;
+use rand_chacha::ChaChaRng;
+use rand_core::{RngCore, SeedableRng};
+
+use crate::{
+    constants::INITIAL_PATH_LENGTH, 
+    state::PlayerPath
+};
+
+pub fn generate_player_path(
+    player: &AccountInfo,
+    player_path: &mut Account<PlayerPath>,
+    bump: u8,
+) -> Result<()> {
+    let seed = player.key().to_bytes();
+    let mut rng = ChaChaRng::from_seed(seed);
```

Comment:
Using ChaChaRng::from_seed(seed) with the player's public key as the seed will always generate the same sequence of random numbers for the same player, because:
- The player's public key (used as seed) never changes
- ChaCha20 is a deterministic random number generator - same seed always produces the same sequence

Something like this would do the job, although it's still pseudo-random until we maybe implement an Oracle VRF:
```suggestion
// Use recent block hash as an additional source of randomness
    let recent_slothash = Clock::get()?.slot_hash.unwrap_or_default();
    
    // Combine player key and recent slot hash for true randomness
    let mut combined_seed = player.key().to_bytes().to_vec();
    combined_seed.extend_from_slice(&recent_slothash.to_bytes());
    
    // Hash the combined seed to get a fixed-length seed
    let seed = hash::hash(&combined_seed).to_bytes();
    let mut rng = ChaChaRng::from_seed(seed);
```